client/my-sites/stats/stats-notices/index.tsx

```diff
@@ -96,9 +96,12 @@ const NewStatsNotices = ( { siteId, isOdysseyStats, statsPurchaseSuccess }: Stat
 		hasSiteProductJetpackStatsPWYWOnly( state, siteId )
 	);
 
-	const shouldShowPaywallNotice = useSelector( ( state ) => {
-		return hasReachedPaywallMonthlyViews( state, siteId );
-	} );
+	// Show the paywall notice if the site has reached the monthly views limit
+	// and no commercial purchase.
+	const shouldShowPaywallNotice =
+		useSelector( ( state ) => {
+			return hasReachedPaywallMonthlyViews( state, siteId );
+		} ) && ! isCommercialOwned;
```

Comment:
Sorry I didn't mention this in my previous comment - maybe we want to check commercial use here, which has Complete and Professional plans checked in there too?

Reply:
Indeed, we should use `supportCommercialUse` in most cases. 😅 Fixed in [bdb587d](https://github.com/Automattic/wp-calypso/pull/92724/commits/bdb587d0ab2ba1fe21e7d185c65e548b2acb81d5).